src/WTG.Analyzers/Analyzers/LinqEnumerable/LinqEnumerableAnalyzer.cs

```diff
@@ -52,189 +55,205 @@ public static void CompilationStart(CompilationStartAnalysisContext context)
 				return;
 			}
 
+			/* exclusively checking the existence of the Enumerable.Append method because 
+			   Enumerable.Append and Enumerable.Prepend were released at the same time */
 			var hasEnumerableAppendMethod = HasEnumerableAppendMethod(context.Compilation);
 			var cache = new FileDetailCache();
 
 			context.RegisterSyntaxNodeAction(c => Analyze(c, cache, hasEnumerableAppendMethod), SyntaxKind.SimpleMemberAccessExpression);
 		}
 
-		static bool ContainsSingleElement (ExpressionSyntax? e)
-		{
-			switch (e?.Kind())
-			{
-				case SyntaxKind.ImplicitArrayCreationExpression:
-					return ((ImplicitArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ObjectCreationExpression:
-					return ((ObjectCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				case SyntaxKind.ArrayCreationExpression:
-					return ((ArrayCreationExpressionSyntax)e)?.Initializer?.Expressions.Count == 1;
-				default:
-					return false;
-			}
-		}
+		static bool ContainsSingleElement(ExpressionSyntax? e) => LinqEnumerableUtils.GetInitializer(e)?.Expressions.Count == 1;
 
 		public static bool LooksLikeAppend(InvocationExpressionSyntax invocation) => (invocation.ArgumentList.Arguments.Count == 1 ? ContainsSingleElement(invocation.ArgumentList.Arguments[0].Expression) : ContainsSingleElement(invocation.ArgumentList.Arguments[1].Expression));
 
 		public static bool LooksLikePrepend(InvocationExpressionSyntax invocation)
 		{
-			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
+			var expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression.TryGetExpressionFromParenthesizedExpression();
 
-			if (expression.IsKind(SyntaxKind.ParenthesizedExpression))
+			if (expression == null)
 			{
-				expression = ((ParenthesizedExpressionSyntax)expression).GetExpression();
+				expression = ((MemberAccessExpressionSyntax)invocation.Expression).Expression;
 			}
 
 			return (invocation.ArgumentList.Arguments.Count == 1 ? ContainsSingleElement(expression) : ContainsSingleElement(invocation.ArgumentList.Arguments[0].Expression));
 		}
 
-		public static bool IsList(SemanticModel model, ObjectCreationExpressionSyntax e) => e.Type.ToString().StartsWith("List", StringComparison.Ordinal) && model.GetTypeInfo(e).Type?.MetadataName == "List`1";
-		public static bool IsIEnumerable(SemanticModel model, ExpressionSyntax e) => model.GetTypeInfo(e).Type?.MetadataName == "IEnumerable`1";
+		public static bool IsMonadicArityCollection (SemanticModel model, ExpressionSyntax e)
```

Comment:
Pulling out the fancy words? I would prefer IsSingleArityCollection, which lines up with other APIs like .Single() or .SingleOrDefault().

Reply:
I was just like "I know there's a word for this in math... una... no... mon... monadic? monadic? monadic." But, I'll change that to SingleArity because it's just more intelligible.